src/s3.py

```diff
@@ -29,30 +33,36 @@ def create_if_not_exists(self):
         bucket does not already exists. Fails if bucket is created within
         same AWS account or another AWS account
         '''
+
         if self.bucket in self.s3_res.buckets.all():
             raise ValueError("A bucket named '{}' already exists under your current AWS account.".format(self.bucket_name))
         else:
             try:
                 self.s3.create_bucket(Bucket=self.bucket_name,
-                            CreateBucketConfiguration={'LocationConstraint': self.region},)
+                                      CreateBucketConfiguration={'LocationConstraint': self.region})
                 print ("Bucket '{}' created.".format(self.bucket_name))
             except ClientError as e:
                 if e.response['Error']['Code'] == 'BucketAlreadyExists':
                     raise ValueError("Failed to create bucket '{}': already exists under separate AWS account. Try a different bucket name.".format(self.bucket_name))
                 else:
                     print(e)
-    
+
+    def get_bucket_name(self):
+        return self.bucket_name
+
     def upload_model(self, model, model_name):
         '''
         Uploades a trained model to the S3 bucket, if a model with the
         specified name does not already exists.
         '''
+
         self.s3.put_object(Bucket=self.bucket_name, Key=model_name, Body=model)
 
     def upload_function(self, function, function_name):
         '''
         Uploades a Lambda function to the S3 bucket, if a function with the
         specified name does not already exists.
         '''
-        self.s3.put_object(Bucket=self.bucket_name, Key=model_name, Body=model)
-
+        # I believe we specify the absolute path instead of the file?
+        # validate functiion name in AWS lambda
+        self.s3.put_object(Bucket=self.bucket_name, Key=function_name, Body=function)
```

Comment:
I think both a filename and an abs path will work in this scenario. The only thing we will need to revisit is the upload_model function's s3.put_object() call since it will take in a byte stream, and there's special handling for that case.

Reply:
does it require a file path or a file object?

Reply:
You can either choose to specify a file path, or pass through a byte stream. This means, if we choose to pass in an object, we will have to convert it into a byte stream so it's properly uploaded into S3. In the case of the function, we should expect the user to create a .py file containing the function code, so we should direct them to pass through the file path in the class that will be exposed to them. 

In the case of the model, we'll have to work together to determine how to properly convert the object into a stream, since we're planning on serializing the model.